app-text/mupdf/mupdf-1.19.1.recipe

```diff
@@ -0,0 +1,135 @@
+SUMMARY="A lightweight XPS and PDF rendering library"
+DESCRIPTION="The renderer in MuPDF is tailored for high quality anti-aliased \
+graphics. It renders text with metrics and spacing accurate to within \
+fractions of a pixel for the highest fidelity in reproducing the look of a \
+printed page on screen.
+
+MuPDF is also small, fast, and yet complete. It supports PDF 1.7 with \
+transparency, encryption, hyperlinks, annotations, searching and more. It also \
+reads XPS/OpenXPS documents and CBZ (Comic Book archive) files."
+HOMEPAGE="https://www.mupdf.com/"
+COPYRIGHT="2006-2022 Artifex Software, Inc"
+LICENSE="AGPL-3.0"
+REVISION="1"
+SOURCE_URI="https://mupdf.com/downloads/archive/mupdf-$portVersion-source.tar.xz"
+CHECKSUM_SHA256="b5eac663fe74f33c430eda342f655cf41fa73d71610f0884768a856a82e3803e"
+SOURCE_DIR="mupdf-$portVersion-source"
+PATCHES="mupdf-$portVersion.patchset"
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="x86"
+
+commandSuffix=$secondaryArchSuffix
+commandBinDir=$binDir
+if [ "$targetArchitecture" = x86_gcc2 ]; then
+	commandSuffix=
+	commandBinDir=$prefix/bin
+fi
+
+# get the library version number (excluding patch rev) from port version
+libVersion="$portVersion"
+libVersionCompat="$libVersion compat >= ${libVersion%.**}"
```

Comment:
Checked with a local build here and with python3.7 and python3.10 works OK for pymupdf:
```diff
-# get the library version number (excluding patch rev) from port version
-libVersion="$portVersion"
-libVersionCompat="$libVersion compat >= ${libVersion%.**}"
+portVersionCompat="$portVersion compat >= ${portVersion%.**}"
```

Reply:
For this I changed requirement for pymupdf to use >= 1.19 for the devel:libmupdf and seems to work fine ;)

Reply:
$portVersion for pymupdf isn't the same as mupdf ;)
```
devel:libmupdf$secondaryArchSuffix >= 1.19
```
Should work

Reply:
$portVersion isn't the same but ${portVersion%.**} should be the same, because the major and minor versions of pymupdf track those of mupdf exactly. Will try what you suggested anyway, cheers

Reply:
Reminder, that is for pymupdf, not for mupdf

Reply:
On further examination I think this failed because I hadn't set the version for devel:libmupdf properly in the recipe for mupdf. Will continue this at #6824